revolv/payments/tests/models/test_payment.py

```diff
@@ -56,46 +70,169 @@ def test_donations(self):
         project2 = Project.factories.base.create()
 
         self._create_payment(user1).save()
-        self._create_payment(user1, instrument_type=self.reinvestment).save()
-        self._create_payment(user1, instrument_type=self.repayment).save()
-
-        self._create_payment(user2, instrument_type=self.repayment).save()
+        self._create_payment(user1, payment_type=self.reinvestment).save()
 
         self.assertEquals(Payment.objects.donations(user1).count(), 1)
         self.assertEquals(Payment.objects.donations(user1, project2).count(), 0)
 
-        self._create_payment(user1, project2).save()
+        self._create_payment(user1, project=project2).save()
         self.assertEquals(Payment.objects.donations(user1, project2).count(), 1)
 
         self.assertEquals(Payment.objects.donations(user2).count(), 0)
 
-    def test_reinvestments(self):
-        """Test that we can bookkeep reinvestment information."""
+    def test_proportion_donated(self):
+        """Verify that repayment proportions are correct."""
+        user1, user2, admin1 = RevolvUserProfile.factories.base.create_batch(3)
+        project = Project.factories.base.create()
+
+        self.assertEquals(Payment.objects.donations(project=project).count(), 0)
+
+        self._create_payment(user1, 30.00, project).save()
+        self.assertEquals(Payment.objects.donations(project=project).count(), 1)
+        self.assertEquals(project.proportion_donated(user1), 1.0)
+
+        self._create_payment(user2, 10.00, project).save()
+        self.assertEquals(Payment.objects.donations(project=project).count(), 2)
+        self.assertEquals(project.proportion_donated(user2), 10.0 / 40.0)
+        self.assertEquals(project.proportion_donated(user1), 30.0 / 40.0)
+
+        self._create_admin_repayment(admin1, 100.00, project).save()
+        self._create_admin_reinvestment(admin1, 100.00, project).save()
+        self.assertEquals(Payment.objects.donations(project=project).count(), 2)
+        self.assertEquals(project.proportion_donated(user2), 10.0 / 40.0)
+        self.assertEquals(project.proportion_donated(user1), 30.0 / 40.0)
+
+    def test_repayment(self):
+        """Test repayment bookkeeping."""
+        admin1, admin2 = RevolvUserProfile.factories.base.create_batch(2)
+        admin1.make_administrator()
+        admin2.make_administrator()
+
+        user1, user2 = RevolvUserProfile.factories.base.create_batch(2)
+        project1, project2 = Project.factories.base.create_batch(2)
+
+        self._create_payment(user1, amount=10.00, project=project1).save()
+        self._create_payment(user2, amount=30.00, project=project1).save()
+        project1.complete_project()
```

Comment:
Should we also assert that a project that is NOT completed cannot have admin repayments made on it?